guides/web-apps/acceptance-testing-with-selenium/README.md

```diff
@@ -114,157 +84,61 @@ We don't want to test only Firefox. Because of that, you should use Selenium Rem
 
-What this means is that you need to change every instance of :
+Open your <code>Properties</code> in the <code>Tests</code> project. Go to <code>Reference Paths</code>, enter C:\Selenium and click Add Folder.
 
-```cs
-[TestFixture]
-```
+### Use BaseTest class to run tests in multiple browsers
 
-To be:
+BaseTest is a helper class that makes it easier to test multiple browsers. The source code is available:
 
-```cs
-[TestFixture("firefox")]
-[TestFixture("chrome")]
-[TestFixture("edge")]
-```
+- BaseTest helper class [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/BaseTest.cs)
+- Using of BaseTest class [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionBoolean/CheckboxPageTest.cs)
 
-Finally, add this constructor somewhere on top of the class. It stores the fixture parameter in the `this.browser` variable:
+When you rebuild the test project now, you should see each test for every browser.
 
+The final setup looks like this:
 
-```cs
-private string browser;
+![visual studio selenium screenshot](/assets/2016-04-01-13_51_26-Launcher-Microsoft-Visual-Studio.png)
 
-public PageLoads(string browser)
-{
-    this.browser = browser;
-}
-```
+Before you can execute the tests, start Selenium Server Standalone by calling `java -jar selenium-server-standalone-3.*.jar`.
 
-When you rebuild the test project now, you should see 3 tests in the Test Explorer, each test for every browser.
+## Wait for asynchronous content
 
-The final setup looks like this:
+There is one common pitfall when writing Selenium tests. The test is executed with disregard of the asynchronously loaded content. This means that your tests need to explicitly wait for UI elements to appear before running any actions or assertions on them.
 
-![visual studio selenium screenshot](/assets/2016-04-01-13_51_26-Launcher-Microsoft-Visual-Studio.png)
+It is a good practice to always wait:
 
-Before you can execute the tests, start Selenium Server Standalone by calling `java -jar selenium-server-standalone-3.0.0-beta3.jar`.
+- Wait for a text element to be present before you check the content of that element
 
-Now, press "Run All" and relax. This will take a while as the system runs your test in three different browsers!
+	- An **example** can be found [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs)	
+	- Please take a look at [TextareaPage_WriteToTextArea()](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs#L29) test and [line 37](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs#L37).	
+	- Method `WaitForText()` is use to wait for text in `TextareaInfoLabel` for maximum 5 second. Then assert is true and test pass.
```

Comment:
I would rather read the above as:

- An **example** can be found in the method `TextareaPage_WriteToTextArea` in KitchenSink Tests (see [TextareaPageTest.cs lines 28-38](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs#L28-L38)). The method `WaitForText()` is used to compare the text value of `TextareaInfoLabel` asynchronously. The assertion passes if the text is found within 5 seconds, otherwise it fails.